services/wallet/api.go

```diff
@@ -574,7 +581,28 @@ func (api *API) SendTransactionWithSignature(ctx context.Context, chainID uint64
 
 func (api *API) CreateMultiTransaction(ctx context.Context, multiTransactionCommand *transfer.MultiTransactionCommand, data []*bridge.TransactionBridge, password string) (*transfer.MultiTransactionCommandResult, error) {
 	log.Debug("[WalletAPI:: CreateMultiTransaction] create multi transaction")
-	return api.s.transactionManager.CreateMultiTransactionFromCommand(ctx, multiTransactionCommand, data, api.router.bridges, password)
+
+	selectedAccount, err := api.getVerifiedWalletAccount(multiTransactionCommand.FromAddress.Hex(), password)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd, err := api.s.transactionManager.CreateMultiTransactionFromCommand(ctx, multiTransactionCommand, data)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO Store transaction in DB AFTER it is sent. This needs taking care of keycard signing case
+	_, err = api.s.transactionManager.InsertMultiTransaction(cmd)
```

Comment:
@IvanBelyakoff would be really good if we could add multi tx to the db once we're sure that submitting the tx was successfully done, otherwise we may have entries in the DB for txs that are not placed ever.